node/server.go

```diff
@@ -348,11 +353,42 @@ func Run(ctx context.Context, opts Options) error {
 		}
 	}()
 
+	// this is not used by cmux since it's listening for browsers on port 443
+	go serveHTTPS()
+
 	<-ctx.Done()
 
 	return ctx.Err()
 }
 
+func serveHTTPS() {
+	certmagic.DefaultACME.DNS01Solver = &certmagic.DNS01Solver{
+		DNSProvider: &cloudflare.Provider{
+			APIToken: "api_token",
+		},
+	}
+
+	u, err := url.Parse("ws://localhost:41505")
+	if err != nil {
+		panic(err)
+	}
+
+	wsProxy := websocketproxy.NewProxy(u)
+	wsProxy.Upgrader = &websocket.Upgrader{
+		ReadBufferSize:  1024,
+		WriteBufferSize: 1024,
+		CheckOrigin: func(r *http.Request) bool {
+			return true
+		},
+		EnableCompression: false,
+	}
+
+	err = certmagic.HTTPS([]string{"url"}, wsProxy)
+	if err != nil {
+		panic(err)
+	}
+}
+
```

Comment:
@tchardin I was thinking of adding a new question in the survey to set the provider api key when a Facilitator starts a node and the url to listen on for providers (a facilitator is also a provider) 

So basically 3 possibilities, a node can be : 
- A Facilitator : needs the provider's (Cloudflare, AWS, ...) api_token + the url of the domain to listen on : "myel.zone"
- A Provider : needs only the subdomain name  "provider1"
- A Client : nothing needed since it won't listen for wss connections